components/validation/src/main/java/org/wso2/carbon/identity/x509Certificate/validation/internal/X509CertificateValidationServiceComponent.java

```diff
@@ -108,4 +111,78 @@ protected void unsetRealmService(RealmService realmService) {
         CertValidationDataHolder.getInstance().setRealmService(null);
     }
 
+    /**
+     * Set the ConfigurationManager.
+     *
+     * @param configurationManager The {@code ConfigurationManager} instance.
+     */
+    @Reference(
+            name = "resource.configuration.manager",
+            service = ConfigurationManager.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unregisterConfigurationManager"
+    )
+    protected void registerConfigurationManager(ConfigurationManager configurationManager) {
+
+        log.debug("Registering the ConfigurationManager in Certificate Validation Service Component.");
+        CertValidationDataHolder.getInstance().setConfigurationManager(configurationManager);
+    }
+
+    /**
+     * Unset the ConfigurationManager.
+     *
+     * @param configurationManager The {@code ConfigurationManager} instance.
+     */
+    protected void unregisterConfigurationManager(ConfigurationManager configurationManager) {
+
+        log.debug("Unregistering the ConfigurationManager in Certificate Validation Service Component.");
+        CertValidationDataHolder.getInstance().setConfigurationManager(null);
+    }
+
+    /**
+     * Set the CertificateManagementService.
+     *
+     * @param certificateManagementService The {@code CertificateManagementService} instance.
+     */
+    @Reference(
+            name = "certificate.mgt.service.component",
+            service = CertificateManagementService.class,
+            cardinality = ReferenceCardinality.MANDATORY,
+            policy = ReferencePolicy.DYNAMIC,
+            unbind = "unsetCertificateManagementService")
+    protected void setCertificateManagementService(CertificateManagementService certificateManagementService) {
+
+        CertValidationDataHolder.getInstance().setCertificateManagementService(certificateManagementService);
```

Comment:
Lets add the debug logs in consistent manner.

Reply:
fixed with https://github.com/wso2-extensions/identity-x509-commons/pull/47/commits/9a05e8b94621f18ecf274b5c8c4930430db2dcec